float64 dist
float64 x_cord
float64 y_cord
float64 radius